electrumsv/wallet.py

```diff
@@ -1346,24 +1355,25 @@ def sign_transaction(self, tx: Transaction, password: str) -> None:
             self.add_transaction(tx_hash, tx, TxFlags.StateSigned)
 
     def get_payment_status(self, req: PaymentRequestRow) -> Tuple[bool, int]:
-        local_height = self._wallet.get_local_height()
-        related_utxos = [ u for u in self._utxos.values()
-            if u.keyinstance_id == req.keyinstance_id ]
-        l = []
-        for utxo in related_utxos:
-            tx_height = self._wallet._transaction_cache.get_height(utxo.tx_hash)
-            if tx_height is not None:
-                confirmations = local_height - tx_height
-            else:
-                confirmations = 0
-            l.append((confirmations, req.value))
-
-        vsum = 0
-        for conf, v in reversed(sorted(l)):
-            vsum += v
-            if vsum >= req.value:
-                return True, conf
-        return False, None
+        with self._utxos_lock:
```

Comment:
Is there any value in locking the whole function over just locking the utxos access?

Reply:
No particular reason. I will make it more fine-grained.

Reply:
pushed.